src/scene/layer.js

```diff
@@ -55,20 +55,35 @@ class InstanceList {
         this.transparentMeshInstances = [];
         this.shadowCasters = [];
 
-        // arrays of VisibleInstanceList for each camera
+        // arrays of VisibleInstanceList for each camera of this layer
         this.visibleOpaque = [];
         this.visibleTransparent = [];
     }
 
-    clearVisibleLists(cameraPass) {
-        if (this.visibleOpaque[cameraPass]) {
-            this.visibleOpaque[cameraPass].length = 0;
-            this.visibleOpaque[cameraPass].list.length = 0;
+    // prepare for culling of camera with specified index
+    prepare(index) {
+
+        // make sure visibility lists are allocated
+        if (!this.visibleOpaque[index]) {
+            this.visibleOpaque[index] = new VisibleInstanceList();
         }
 
-        if (this.visibleTransparent[cameraPass]) {
-            this.visibleTransparent[cameraPass].length = 0;
-            this.visibleTransparent[cameraPass].list.length = 0;
+        if (!this.visibleTransparent[index]) {
+            this.visibleTransparent[index] = new VisibleInstanceList();
+        }
+
+        // mark them as not processed yet
+        this.visibleOpaque[index].done = false;
+        this.visibleTransparent[index].done = false;
+    }
+
+    // delete entry for a camera with specified index
+    delete(index) {
+        if (index < this.visibleOpaque.length) {
+            this.visibleOpaque.splice(index, 1);
```

Comment:
not sure how this list is used, but just wanted to check that splicing the array (and so changing the indexes of the following elements) is ok?

Reply:
yep, these are just caches of culled mesh instances, and get updated every frame .. so this is just removing entry from cache to release memory, but no index depends on this